server.js

```diff
@@ -14,124 +14,100 @@ const { parse } = require('url')
-          pathname.substring(1)
-      })
-      res.end()
-    } else if (/^\/(help|chat)\/?$/i.test(pathname)) {
-      // path /(help|chat) -> Discord chat invite
-      res.writeHead(303, { Location: 'https://discordapp.com/invite/dvwXA2N' })
-      res.end()
-    } else if (/^\/(docs|documentation)(\/.*)?$/i.test(pathname)) {
-      // path /docs... or /documentation... -> /doc...
-      res.writeHead(301, {
-        Location: req.url.replace(/\/(docs|documentation)/i, '/doc')
-      })
-      res.end()
-    } else if (/^\/doc\/commands-reference(\/.*)?$/i.test(pathname)) {
-      // path /doc/commands-reference... -> /doc/command-reference...
-      res.writeHead(301, {
-        Location: req.url.replace(
-          '/doc/commands-reference',
-          '/doc/command-reference'
-        )
-      })
-      res.end()
-    } else if (/^\/doc\/tutorial\/?$/i.test(pathname)) {
-      // path /doc/tutorial -> /doc/tutorials
-      res.writeHead(301, {
-        Location: req.url.replace(/\/doc\/tutorial\/?/, '/doc/tutorials')
-      })
-      res.end()
-    } else if (/^\/doc\/tutorial\/(.*)?/.test(pathname)) {
-      // path /doc/tutorial/... -> /doc/tutorials/deep/...
-      res.writeHead(301, {
-        Location: req.url.replace('/doc/tutorial/', '/doc/tutorials/deep/')
-      })
-      res.end()
-    } else if (
-      /^\/doc\/use-cases\/data-and-model-files-versioning\/?$/.test(pathname)
-    ) {
-      // path /doc/use-cases/data-and-model-files-versioning
-      //  ->  /doc/use-cases/versioning-data-and-model-files
-      res.writeHead(301, {
-        Location: req.url.replace(
-          '/doc/use-cases/data-and-model-files-versioning',
-          '/doc/use-cases/versioning-data-and-model-files'
-        )
-      })
-      res.end()
-    } else if (/^\/doc(\/.*)?$/.test(pathname)) {
+const processRedirectString = redirectString => {
+  let [regex, replace, code = 301] = redirectString.split(/\s+/g)
+  const matchPathname = /^\^?\//.test(regex)
+  regex = new RegExp(regex)
+  code = Number(code)
+  return {
+    regex,
+    matchPathname,
+    replace,
+    code
+  }
+}
+
+exports.processRedirectString = processRedirectString
+
+// Parse redirects when starting up
+redirects = redirects.map(processRedirectString)
+
+const matchRedirectList = (host, pathname) => {
+  const wholeUrl = `https://${host}${pathname}`
+
+  for (const { matchPathname, regex, replace, code } of redirects) {
+    const matchTarget = matchPathname ? pathname : wholeUrl
+    if (regex.test(matchTarget)) {
+      return [code, matchTarget.replace(regex, replace)]
+    }
+  }
+
+  return []
+}
+
+const getRedirect = (host, pathname, { req, dev } = {}) => {
+  if (req != null && req.headers['x-forwarded-proto'] !== 'https' && !dev) {
+    return [301, `https://${host.replace(/^www\./, '')}${req.url}`]
+  }
+
+  return matchRedirectList(host, pathname)
+}
+
+exports.getRedirect = getRedirect
+
+// Do not run the server if the module is being required by something else
```

Comment:
let's move this logic to a separate file? redirets.js?

Reply:
so that here we keep only clean simple server runner

Reply:
Sure!